base_local_planner/src/odometry_helper_ros.cpp

```diff
@@ -77,6 +96,65 @@ void OdometryHelperRos::getRobotVel(tf::Stamped<tf::Pose>& robot_vel) {
   robot_vel.stamp_ = ros::Time();
 }
 
+void OdometryHelperRos::getEstimatedRobotVel(tf::Stamped<tf::Pose>& robot_vel) {
```

Comment:
It does make more sense to put it in the odometry node (perhaps publishing on a separate topic).  I put it in the odometry helper because I wanted to eliminate all possible latency and do the estimate directly before it is required.  However, that requires more forking and adding new parameters to move_base.  Wrapping it in the odom node should work.